ambry-tools/src/main/java/com.github.ambry.tools/admin/ConcurrencyTestTool.java

```diff
@@ -0,0 +1,946 @@
+    private final ClusterMap clusterMap;
+    private final int maxBlobSize;
+    private final int minBlobSize;
+    private final boolean enableVerboseLogging;
+    private final ThreadLocalRandom localRandom;
+    private final NetworkClientUtils networkClientUtils;
+    private final AtomicInteger correlationIdGenerator = new AtomicInteger(0);
+
+    public ServerPutGetHelper(Properties properties, String hostName, int port, ClusterMap clusterMap, int maxBlobSize,
+        int minBlobSize, Boolean enableVerboseLogging)
+        throws Exception {
+      VerifiableProperties verifiableProperties = new VerifiableProperties(properties);
+      NetworkClientFactory networkClientFactory =
+          new NetworkClientFactory(new NetworkMetrics(new MetricRegistry()), new NetworkConfig(verifiableProperties),
+              null, 20, 20, 10000, SystemTime.getInstance());
+      NetworkClient networkClient = networkClientFactory.getNetworkClient();
+      this.networkClientUtils = new NetworkClientUtils(networkClient, 50);
+      this.hostName = hostName;
+      this.port = port;
+      this.clusterMap = clusterMap;
+      this.maxBlobSize = maxBlobSize;
+      this.minBlobSize = minBlobSize;
+      this.enableVerboseLogging = enableVerboseLogging;
+      this.localRandom = ThreadLocalRandom.current();
+    }
+
+    /**
+     * {@inheritDoc}
+     * Uploads a blob to the server directly and returns the blobId along with the content
+     * @return a {@link Pair<String, byte[]>} which contains the blobId and the content
+     * @throws IOException
+     */
+    @Override
+    public FutureResult produce(FutureResult futureResult, Callback callback) {
+      int randomNum = localRandom.nextInt((maxBlobSize - minBlobSize) + 1) + minBlobSize;
+      byte[] blob = new byte[randomNum];
+      byte[] usermetadata = new byte[new Random().nextInt(1024)];
+      BlobProperties props = new BlobProperties(randomNum, "test");
+      int correlationId = -1;
+      try {
+        List<PartitionId> partitionIds = clusterMap.getWritablePartitionIds();
+        int index = (int) getRandomLong(localRandom, partitionIds.size());
+        PartitionId partitionId = partitionIds.get(index);
+        BlobId blobId = new BlobId(partitionId);
+        correlationId = correlationIdGenerator.incrementAndGet();
+        PutRequest putRequest =
+            new PutRequest(correlationId, "ConcurrencyTest", blobId, props, ByteBuffer.wrap(usermetadata),
+                ByteBuffer.wrap(blob), props.getBlobSize(), BlobType.DataBlob);
+        long startTime = SystemTime.getInstance().nanoseconds();
+        List<NetworkClientUtils.RequestMetadata> requestMetadataList = new ArrayList<>();
+        NetworkClientUtils.RequestMetadata requestMetadata = new NetworkClientUtils.RequestMetadata<>();
+        requestMetadata.requestInfo = new RequestInfo(hostName, new Port(port, PortType.PLAINTEXT), putRequest);
+        requestMetadata.futureResult = new FutureResult<ByteBuffer>();
```

Comment:
Sure. I have fixed at places that I knew of. If you find anything more(after I update this PR), can you be bit specific.